lib/src/revset.rs

```diff
@@ -2660,6 +2678,7 @@ pub struct RevsetParseContext<'a> {
     user_email: String,
     extensions: &'a RevsetExtensions,
     workspace: Option<RevsetWorkspaceContext<'a>>,
+    function_map: HashMap<&'static str, &'a RevsetFunction>,
```

Comment:
nit: unless we plan to turn `RevsetFunction` into `Box<dyn Fn>`, `RevsetFunction` can be just copied.